tools/sdk/build_rust_sdk.sh

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+
+# Exit on error
+set -e
+
+# Ask to build from local source or to clone the repository
+read -p "Do you want to build the Rust SDK from local source (yes/no) default to yes? " buildLocal
+buildLocal=${buildLocal:-yes}
+
+date=$(gdate +%Y%m%d%H%M%S)
+
+# Ask for the Rust SDK local source path
+# if folder rustSdk/ exists, use it as default
+if [ ${buildLocal} == "yes" ]; then
+    read -p "Please enter the path to the Rust SDK local source, default to ../matrix-rust-sdk" rustSdkPath
+    rustSdkPath=${rustSdkPath:-../matrix-rust-sdk/}
+    if [ ! -d "${rustSdkPath}" ]; then
+        printf "\nFolder ${rustSdkPath} does not exist. Please clone the matrix-rust-sdk repository in the folder ../matrix-rust-sdk.\n\n"
+        exit 0
+    fi
+else
+    read -p "Please enter the Rust SDK repository url, default to https://github.com/matrix-org/matrix-rust-sdk.git " rustSdkUrl
+    rustSdkUrl=${rustSdkUrl:-https://github.com/matrix-org/matrix-rust-sdk.git}
+    read -p "Please enter the Rust SDK branch, default to main " rustSdkBranch
+    rustSdkBranch=${rustSdkBranch:-main}
+    cd ..
+    git clone ${rustSdkUrl} matrix-rust-sdk-$date
+    cd matrix-rust-sdk-$date
+    git checkout ${rustSdkBranch}
+    rustSdkPath=$(pwd)
+    cd ../element-x-android
+fi
+
+
+cd ${rustSdkPath}
+git status
+
+read -p "Will build with this version of the Rust SDK ^. Is it correct (yes/no) default to yes? " sdkCorrect
+sdkCorrect=${sdkCorrect:-yes}
+
+if [ ${sdkCorrect} != "yes" ]; then
+    exit 0
+fi
+
+# Ask if the user wants to build the app after
+read -p "Do you want to build the app after (yes/no) default to yes? " buildApp
+buildApp=${buildApp:-yes}
+
+# If folder ../matrix-rust-components-kotlin does not exist, close the repo
+if [ ! -d "../matrix-rust-components-kotlin" ]; then
+    printf "\nFolder ../matrix-rust-components-kotlin does not exist. Cloning the repository into ../matrix-rust-components-kotlin.\n\n"
+    git clone https://github.com/matrix-org/matrix-rust-components-kotlin.git ../matrix-rust-components-kotlin
+fi
+
+printf "\nResetting matrix-rust-components-kotlin to the latest main branch...\n\n"
+cd ../matrix-rust-components-kotlin
+git reset --hard
+git checkout main
+git pull
+
+printf "\nBuilding the SDK...\n\n"
+./scripts/build.sh -p ${rustSdkPath} -m sdk -t aarch64-linux-android -o ../element-x-android/libraries/rustsdk
```

Comment:
Won't this only work for ARM64 emulators/devices? Maybe the user who builds this wants to use an x86-64 emulator in Linux/Windows.

Reply:
You could use `uname -m` to get the architecture and invoke the correct option.
We could reasonably assume, being this for local use, crosscompilation is not a goal.

So if `uname -m` is x86_64 we build the x86_64 version.
If `uname -m` is amd64 we build the aarch64 version.

Reply:
More hints at: https://askubuntu.com/questions/803596/cpu-architecture-scanning-using-bash